packages/design-system/src/styles/utilities/_ring.scss

```diff
@@ -0,0 +1,14 @@
+@use 'sass:math';
+
+.ring,
+.focus\:ring:focus,
+.focus-within\:ring:focus-within,
+.focus-visible\:ring:focus-visible {
```

Comment:
While this method makes it flexible enough to use anywhere, I don't actually think we have a use case at the moment for applying it to unfocused elements, and I don't think we want to encourage adding and removing the classes programmatically when the `:focus-within` state should cover pretty much everything. Even though having the extra class doesn't harm anything, it does make it tricky to name. I would actually think this class is more discoverable if it's named "focus" instead of "ring", but the word "ring" is needed if "focus" is taken by ":focus". What about the following names?
- `ds-u-focus` (corresponds to `:focus`)
- `ds-u-focus-witin` (corresponds to `:focus-within`)
- `ds-u-focus-visible` (corresponds to `:focus-visible`) (and adding `:focus-visible` was a good idea 👍)

Reply:
@line47, do you have an opinion on the utility class naming?

Reply:
The only instance I could think of where a generic `.ring`/`.focus` class would be useful was in instances we were applying focus on non-interactive elements (I think we did this with heading text in Drawer and Dialog), but that pattern seems to be getting deprecated, so I agree removing it's a good idea.